the-collab-lab/tcl-66-smart-shopping-list
# Ec issue #29 bug fix - Lists in the Nav Bar throw error when local storage is empty

## Description
The list elements in the Navigation Bar contained a const declaration that used the "getItem" method to obtain a value from local storage. When this method was run, an error would occur when local storage was empty or did not contain a value for the key, "tcl-shopping-list-path". Specifically, the application would throw an error and the home page would be blank after signing in to the app. 

## Updates

To resolve this bug that stemmed from the `NavigationBarSingleList` component, I changed the localStorageListName const declaration to an arrow function and added a try-catch statement.

### Before Bug Fix
![image](https://github.com/the-collab-lab/tcl-66-smart-shopping-list/assets/105468537/db0dc48f-b7c1-43a5-bdab-fba8d9e5e273)

## Testing Steps / QA Criteria
Steps to reproduce the behavior:
 * Git pull
 * Open the branch named ec-issue-29-bug-fix
 * Run npm start
 * Sign out of the application by using the Sign Out button in the lower-left hand corner.
 * Open the Inspect Element tool on Chrome and open the "Application tab". Confirm that the local storage is empty/cleared.
 * Sign in to the application. The home page should appear. Confirm that no error is thrown by the application in the console log.
